alert/search.go

```diff
@@ -134,6 +140,12 @@ var searchTemplate = template.Must(template.New("alert-search").Funcs(search.Hel
 			{{ end }}
 		)
 	{{ end }}
+	{{ if not .ClosedBefore.IsZero }}
+		AND (select timestamp from alert_logs where alert_id = a.id and event = 'closed') < :closedBeforeTime
```

Comment:
This should probably be `and exists (select 1 from ...)` (`1` is just a convention for an unused placeholder). Also, since this will largely be used with metrics, I think we should join against the `alert_metrics` table instead of logs so that we don't have an intermediate state where an alert is returned from this filter before metrics are recorded. It should be a cheaper index as well.